File: Binary_Prefix_Divisible_By_5.java
class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        int sum = 0;
        int length = nums.length;
        List<Boolean> result = new ArrayList<>();
        //Go through the nums notice that shifting a binary number to the left is equals to multiplying a decimal number by 2
        for (int i=0; i<length; i++) {
            sum = sum*2 + nums[i];
            if (sum%5==0) {
                result.add(true);
            }
            else {
                result.add(false);
            }
            //This is to prevent it from overflow
            sum = sum%5;
        }
        return result;
    }
}

File: Binary_Prefix_Divisible_By_5.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {\
    
    vector<bool> res;
    int num=0;
    
    //loop over the whole array numbers
    for(int i=0;i<nums.size();i++){
        num=(num*2+nums[i])%5;
        res.push_back(num==0);
    }
    //return the new boolean values array
    return res;
    }
};
